File: Systems/lesson-1/increment-in-place/increment-array-in-place.c
#include <stdio.h>

void increment_in_place(int* numbers, int count) {
    for (int i = 0; i < count; i++) {
        numbers[i] += 1;
    }
}

void print_array(int* numbers, int count) {
    for (int i = 0; i < count; i++) {
        printf("%d\n", numbers[i]);
    }
}

int main() {
    int test_array[3] = { 0, 1, 2 };

    increment_in_place(test_array, 3);
    print_array(test_array, 3);

    return 0;
}

File: Systems/lesson-1/increment-in-place/increment-array-in-place.js

function increment_array_in_place(array) {
    const count = array.length;
    for (let i = 0; i < count; i++) {
        array[i] += 1;
    }
}

function print_array(array) {
    const count = array.length;
    for (let i = 0; i < count; i++) {
        console.log(array[i]);
    }
}

function main() {
    let test_array = [0, 1, 2];
    increment_array_in_place(test_array);
    print_array(test_array);
}

main();
